src/components/CreateCourse/CreateCourse.jsx

```diff
@@ -1,151 +1,204 @@
+  LABEL_TEXT,
+  getAlertText,
+  PARAMETERS,
+} from "./createCourseStrings";
+
+function CreateCourse() {
+  const navigate = useNavigate();
 
-export const CreateCourse = ({ onCourseCreated }) => {
-  const { addCourse, authors, addAuthor, setAuthors } =
-    useCourseAuthorContext();
   const [title, setTitle] = useState("");
   const [description, setDescription] = useState("");
+  const [newAuthor, setNewAuthor] = useState("");
   const [duration, setDuration] = useState("");
-  const [courseAuthors, setCourseAuthors] = useState([]);
-  const [newAuthorName, setNewAuthorName] = useState("");
+  const [authorsList, setAuthorsList] = useState(mockedAuthorsList);
+  const [selectedAuthorsList, setSelectedAuthorsList] = useState([]);
 
-  const handleAddAuthorToCourse = (authorId) => {
-    AddAuthorToCourse(authorId, authors, courseAuthors, setCourseAuthors);
-  };
+  const ALERT_TEXT = getAlertText();
 
-  const handleRemoveAuthorFromSelectList = (authorId) => {
-    RemoveAuthorFromSelectList(
-      authorId,
-      courseAuthors,
-      setCourseAuthors,
-      authors,
-      setAuthors,
+  function addCourseAuthor(author) {
+    setSelectedAuthorsList([...selectedAuthorsList, author]);
+    setAuthorsList((current) =>
+      current.filter((item) => item.name !== author.name),
     );
-  };
-
-  const handleCreateAuthor = () => {
-    CreateAuthor(newAuthorName, authors, addAuthor, setNewAuthorName);
-  };
+  }
 
-  const handleCreateCourse = () => {
-    CreateNewCourse(
-      title,
-      description,
-      duration,
-      courseAuthors,
-      addCourse,
-      onCourseCreated,
+  function deleteCourseAuthor(author) {
+    setAuthorsList([...authorsList, author]);
+    setSelectedAuthorsList((current) =>
+      current.filter((item) => item.name !== author.name),
     );
-  };
+  }
+
+  function createNewAuthor(author) {
+    if (author.length < PARAMETERS.AUTHOR_MIN_LENGTH) {
+      alert(ALERT_TEXT.AUTHOR_NAME_TOO_SHORT);
+      return;
+    }
+    const newAuthor = {
+      id: uuidv4(),
+      name: author,
+    };
+    setAuthorsList([...authorsList, newAuthor]);
+    mockedAuthorsList.push(newAuthor);
+  }
+
+  function isValid() {
+    if (!title) {
+      alert(ALERT_TEXT.TITLE_REQUIRED);
+      return false;
+    }
+    if (!description) {
+      alert(ALERT_TEXT.DESCRIPTION_REQUIRED);
+      return false;
+    }
+    if (description.length < PARAMETERS.DESCRIPTION_MIN_LENGTH) {
+      alert(ALERT_TEXT.DESCRIPTION_TOO_SHORT);
+      return false;
+    }
+    if (duration <= PARAMETERS.DURATION_MIN_LENGTH) {
+      alert(ALERT_TEXT.DURATION_TOO_SHORT);
+      return false;
+    }
+    if (!selectedAuthorsList.length) {
+      alert(ALERT_TEXT.AUTHOR_REQUIRED);
+      return false;
+    }
+    return true;
+  }
+
+  function cancelCourseCreationHandler() {
+    navigate("/courses");
+  }
+
+  function createCourseSubmitHandler() {
+    if (!isValid()) {
+      alert("problem");
```

Comment:
As I can see - this alert doesn't seem to be useful as you're alerting other messages in the `isValid` function. So, this can be simplified to 
```
function createCourseSubmitHandler() {
  if (isValidCreateCoursePayload()) {
   ...
  }
}
```

Reply:
changed